pkg/history/populate_headers.go

```diff
@@ -59,3 +62,10 @@ func RetrieveAndUpdateHeaders(blockChain core.BlockChain, headerRepository datas
 func getBlockRangeString(blockRange []int64) string {
 	return fmt.Sprintf("Backfilling |%v| blocks", len(blockRange))
 }
+
+func getLastBlock(startingBlock, chainHead, validationWindow int64) int64 {
+	if chainHead-validationWindow < startingBlock {
+		return startingBlock
```

Comment:
Trying to think this through - this is basically saying that if our starting block is less than 15 from the chain, then return the starting block, I think. Would this mean that for the first few blocks of a sync, it's possible we'll see the same duplicate header error since the validation window and the missing headers window will overlap? I think it's not a big deal, since it'll be for less than 15 blocks of a sync, but just want to make sure I'm understanding.

Reply:
Yeah I think that's right, although I'm thinking we can probably not worry about that. I think it would only be possible for the `startingBlock`, and only once (since subsequent backfill processes would see that it exists in the DB if we had hit the constraint violation on an attempted insert)